java/vector/src/main/java/org/apache/arrow/vector/ipc/JsonFileWriter.java

```diff
@@ -83,8 +84,10 @@
 import com.fasterxml.jackson.databind.MappingJsonFactory;
 
 /**
- * A writer that converts binary Vectors into a JSON format suitable
+ * A writer that converts binary Vectors into an <em>internal, unstable</em> JSON format suitable
```

Comment:
If this is intended for integration testing, could it be under the test directory?

Reply:
Integration tests are run via a binary, not via JUnit, so they need to end up in something that gets compiled into a JAR. (Also I'd be a little hesitant to break compatibility immediately here.)